content/nomad/v1.11.x/content/docs/deploy/clusters/connect-nodes.mdx

```diff
@@ -214,6 +217,221 @@ consul {
+1. [Generate a Nomad token](#generate-a-nomad-token).
+1. [Generate a client introduction token](#generate-a-client-introduction-token).
+1. [Configure the Nomad agent to use client introduction](#configure-the-nomad-agent).
+1. [Start the Nomad agent](#start-the-nomad-agent).
+1. [Monitor client join failures](#monitor-client-join-failures).
+
+### Create an ACL policy
+
+This example creates the `node:write` policy required to generate tokens.
+
+1. Create a policy file called `client-introduction.hcl`.
+
+   ```hcl
+   node {
+    policy = "write"
+   }
+   ```
+
+1. Create the policy.
+
+   ```shell-session
+   nomad acl policy apply client-introduction client-introduction.hcl
+   ```
+
+### Create an ACL role
+
+This example creates an ACL role called `client-introduction`.
+
+```shell-session
+nomad acl role create --tls-skip-verify -name="client-introduction" \
+-policy="client-introduction"
+```
+
+Output is similar to the this:
+
+```shell-session
+ID           = cf0b4a43-b00f-cc30-b656-b34d66151b04
+Name         = client-introduction
+Description  = <none>
+Policies     = client-introduction
+Create Index = 117
+Modify Index = 117
+```
+
+### Generate a Nomad token
+
+Generate an ACL token from the `client-introduction` role. Nomad uses this
+token to request client introduction tokens. In a production environment, we
+recommend that you use Vault to generate and manage these tokens.
+
+```shell-session
+nomad acl token create -name="client-intro-token-1" \
+-role-name="client-introduction" -type=client -ttl=8h
+```
+
+Output is similar to this:
+
+```shell-session
+Accessor ID  = 8c22a7c0-44f5-044d-ef84-bfa06118faf4
+Secret ID    = d99d678d-426c-330e-74f3-de53a868e2f9
+Name         = client-intro-token-1
+Type         = client
+Global       = false
+Create Time  = 2025-11-18 21:01:44.62075624 +0000 UTC
+Expiry Time  = 2025-11-19 05:01:44.62075624 +0000 UTC
+Create Index = 128
+Modify Index = 128
+Policies     = []
+
+Roles
+ID                                    Name
+cf0b4a43-b00f-cc30-b656-b34d66151b04  client-introduction
+```
+
+### Generate a client introduction token
+
+The following snippets generate a basic token.
+
+<Tabs>
+<Tab heading="CLI">
+
+Use the [`nomad node intro create` command](/nomad/commands/node/intro/create) to generate the client introduction token.
+
+```shell-session
+nomad node intro create  > intro_token.jwt
+```
+
+The `intro-token.jwt` contents are similar to this.
+
+```shell-session
+ "eyJhbGciOiJSUzI1NiIsImtpZCI6IjljZDgy..."
+```
+
+</Tab>
+<Tab heading="API">
+
+Use the [client introduction identity
+endpoint](/nomad/api-docs/acl/identities#create-client-introduction-identity) to
+generate the client introduction token. This example sends an empty JSON object,
+but you may use a JSON payload file to specify parameters such as `NodeName`,
+`NodePool`, and `TTL`.
+
+```shell-session
+curl \
+    --request POST \
+    --header "X-Nomad-Token: d99d678d-426c-330e-74f3-de53a868e2f9" \
```

Comment:
Here's the `client-intro-token-1` ACL token value. So now I have a few questions.

1. Should the CLI instructions have included a step to use the ACL token as well?
2. Are you supposed to create a separate ACL token for each node, and that's why it's `token-1`?
3. Or should you be creating a separate intro token for each client, using the one shared ACL token?

But now I'm confused about the workflow. You create an ACL token so that a client node has permission to contact a server and change policies (which covers creating client introduction tokens). Then the node uses the client introduction token it creates to join the cluster it already has a valid ACL token from? Why the dedicated client token, then, when there's already an ACL token being used?